js/src/components/legacy-plugin-active-notice/index.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, createInterpolateElement } from '@wordpress/element';
+import { Notice } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { sfwData } from '~/constants';
+import './index.scss';
+
+/**
+ * React component that displays a warning notice with link to docs on how to uninstall the legacy plugin
+ * if the legacy plugin is active.
+ *
+ * @return {JSX.Element|null} A warning notice with a link to docs on how to uninstall the legacy plugin, or null if the legacy plugin is not active.
+ */
+const LegacyPluginActiveNotice = () => {
+	const [ isDismissed, setIsDismissed ] = useState( false );
+	const { isLegacyPluginActive } = sfwData;
+	if ( ! isLegacyPluginActive || isDismissed ) {
+		return null;
+	}
+
+	return (
+		<Notice
+			status="warning"
+			isDismissible={ true }
+			onDismiss={ () => setIsDismissed( true ) }
+			className="sfw-legacy-plugin-active-notice"
```

Comment:
We can remove this class name if we remove all the custom styles for the notice. See my comment below.